src/System.Device.Gpio/System.Device.Gpio.csproj

```diff
@@ -42,4 +43,54 @@
       <IsWinMDFile>true</IsWinMDFile>
     </Reference>
   </ItemGroup>
+
+  <PropertyGroup>
+    <NotSupportedSourceFile>$(IntermediateOutputPath)$(MSBuildProjectName).notsupported.cs</NotSupportedSourceFile>
+  </PropertyGroup>
+  <Target Name="GenerateNotSupportedAssemblyForNetStandardSource"
+          Condition="'$(RuntimeIdentifier)' == 'linux'"
+          Inputs="@(ReferencePath);@(IntermediateRefAssembly)"
+          Outputs="$(NotSupportedSourceFile)"
+          AfterTargets="CoreCompile">
+    <!-- Calculate reference path directories without duplicates. -->
+    <ItemGroup>
+      <!-- Build list where dependencies are located. -->
+      <_referencePathDirectoriesWithDuplicates Include="@(ReferencePath->'%(RootDir)%(Directory)'->TrimEnd('\'))" />
+      <!-- strip metadata, removing duplicates -->
+      <_referencePathDirectories Include="%(_referencePathDirectoriesWithDuplicates.Identity)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <GenAPIArgs>"%(IntermediateRefAssembly.Identity)"</GenAPIArgs>
+      <GenAPIArgs>$(GenAPIArgs) --lib-path "@(_referencePathDirectories)"</GenAPIArgs>
+      <GenAPIArgs>$(GenAPIArgs) --out "$(NotSupportedSourceFile)"</GenAPIArgs>
+      <GenAPIArgs Condition="'$(LangVersion)' != ''">$(GenAPIArgs) --lang-version "$(LangVersion)"</GenAPIArgs>
+      <GenAPIArgs>$(GenAPIArgs) --throw "$(GeneratePlatformNotSupportedAssemblyMessage)"</GenAPIArgs>
+      <GenAPIArgs Condition="'$(GeneratePlatformNotSupportedAssemblyWithGlobalPrefix)' == 'true'">$(GenAPIArgs) --global</GenAPIArgs>
+    </PropertyGroup>
+
+    <Exec Command="$(_GenAPICommand) $(GenAPIArgs)" 
+          WorkingDirectory="$(ToolRuntimePath)" />
+
+    <ItemGroup>
+      <FileWrites Include="$(NotSupportedSourceFile)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="SetupEnvironmentForNotSupportedAssembly">
+    <PropertyGroup>
+      <OutDir>$(OutDir)notsupported\</OutDir>
+      <ProjectDepsFilePath>$(OutDir)$(ProjectDepsFileName)</ProjectDepsFilePath>
+    </PropertyGroup>
+    <ItemGroup>
+      <Compile Remove="@(Compile)" />
+      <Compile Include="$(IntermediateOutputPath)..\$(MSBuildProjectName).notsupported.cs" />
+      <CreateDirectory Include="$(OutputPath)notsupported\" />
```

Comment:
Here you use OutputPath and above you use OutDir.  I wonder if you even need to change this since nothing else will be building to a path without a RID?

Reply:
I don't need to change it, but I just like the output better when it is clear to me which assembly is which, today in the artifacts/bin/Debug/System.Device.Gpio/netstandard2.0 folder I have four sub folders which make their content obvious: ref, notsupported, linux, win. If I remove this, then it would look like: ref, linux, win and then have a System.Device.Gpio.dll in that directory which wouldn't be to obvious that is the turd assembly.

That said, I can still change it if you think it's better.